docs/docs/articles/github-actions.md

```diff
@@ -1,135 +1,85 @@
 
 ```yaml
-# Creating test
-- name: Create test
-  id: create_test
-  uses: kubeshop/testkube-docker-action@v1
-  with:
-    command: create
-    resource: test
-    namespace: testkube
-    parameters: "--type k6/script --name testkube-github-action"
-    stdin: "import http from 'k6/http';\nimport { sleep,check } from 'k6';\n\nexport default function () {\n  const baseURI = `${__ENV.TESTKUBE_HOMEPAGE_URI || 'https://testkube.kubeshop.io'}`\n  check(http.get(`${baseURI}/`), {\n    'check testkube homepage home page': (r) =>\n      r.body.includes('Your Friendly Cloud-Native Testing Framework for Kubernetes'),\n  });\n\n\n  sleep(1);\n}\n"
+steps:
+  # Setup Testkube
+  - uses: kubeshop/setup-testkube@v1
+    with:
+      organization: ${{ secrets.TkOrganization }}
+      environment: ${{ secrets.TkEnvironment }}
+      token: ${{ secrets.TkToken }}
+
+  # Use CLI with a shell script
+  - run: |
+      testkube create test --name some-test-name --file path_to_file.json
+      testkube run test some-test-name -f
 ```
 
-## Configuring Your GH Actions for Access to GKE
-
-To obtain set up access to a GKE (Google Kubernetes Engine) from GH (GitHub) actions, please visit the official documentation from GH: <https://docs.github.com/en/actions/deployment/deploying-to-google-kubernetes-engine>.
-
-1. Create a Service Account (SA).
-2. Save it into GH's **Secrets** of the repository.
-3. Run either `Helm` or `Kubectl kubtest` commands against the set up GKE cluster.
-
-## Main GH Action Section Configuration
-
-To install on Linux or MacOS, run:
-
+Create and run tests on self-hosted platform. Please mind that it requires establishing connection with the k8s cluster:
 ```yaml
-# Deploy into configured GKE cluster:
-- name: Deploy
-  run: |-
-    helm upgrade --install --atomic --timeout 180s testkube helm-charts/testkube --namespace testkube --create-namespace
-```
-
-In addition to Helm, you can run any other K8s-native command. In our case: `testkube...`
-
-## Complete Example of Working GH Actions Workflow and Testkube Tests Usage
-
-Testkube tests can be easily re-used with minimal modifications according to your needs.
-
-To run tests on Linux or MacOS:
+steps:
+  - uses: aws-actions/configure-aws-credentials@v4
+    with:
+      aws-access-key-id: ${{ secrets.AwsAccessKeyId }}
+      aws-secret-access-key: ${{ secrets.AwsSecretAccessKey }}
+      aws-region: ${{ secrets.AwsRegion }}
+
+  - run: |
+      aws eks update-kubeconfig --name ${{ secrets.EksClusterName }} --region ${{ secrets.AwsRegion }}
+
+  # Setup Testkube
+  - uses: kubeshop/setup-testkube@v1
+  # Use CLI with a shell script
+  - run: |
+      testkube create test --name some-test-name --file path_to_file.json
+      testkube run test some-test-name -f 
+ ```
+:::note
+Please note that it is required to have Testkube Helm charts deployed into your k8s cluster prior to creating and executing tests. We advise to do this in a separate workflow.
```

Comment:
* Secrets - same as above
* We should split it to (sub-)sections for easier navigation
* The example for setting up Testkube connects to AWS EKS, while deployment doesn't have a connection